---
layout: single
title: "Publications"
permalink: /publications/
author_profile: true
classes: wide
---

<style>

/* bibtext block */
.noshow { display: none; }

.bibtex, .abstract-box {
  margin-top: 1em;
  margin-bottom: 1em;
  border: 1px solid #ccc;
  padding: 1em;
  background: #fdfdfd;
  position: relative;
}

.bibtex pre {
  font-size: 85%;
  overflow: auto;
  white-space: pre-wrap;
  font-family: monospace;
}

 /* abstract box */
.abstract-box {
  font-size: 95%;
  line-height: 1.4em;
}

/* toggle buttons */
.button-group {
  margin-top: 0.1em;
}

.toggle-button, .link-button {
  display: inline-block;
  font-size: 0.8em;
  padding: 4px 10px;
  margin-right: 6px;
  margin-top: 4px;
  background-color: #eee;
  border: none;
  border-radius: 3px;
  cursor: pointer;
  text-decoration: none;
  color: black;
}

.toggle-button:hover, .link-button:hover {
  background-color: #ddd;
}

<style>
  /* ref list custom */
ul.no-bullets {
  list-style: none;
  padding-left: 0;
  margin-left: 0; /* also reset margin */
}

ul.no-bullets li {
  list-style: none;    /* important: remove bullet from li */
  margin-left: 0;      /* reset default li indent */
  padding-left: 0;
  margin-bottom: 0.6em;
}

  /* year blocks */
.year-label {
  position: relative;
  text-align: right;
  margin: 2em 0 1.5em;
  font-weight: normal;
  font-size: 0.9em;
  color: #888;
}

.year-label::before {
  content: "";
  display: block;
  height: 1px;
  background-color: #ccc;
  position: absolute;
  top: 50%;
  left: 0;
  width: 100%;
  z-index: 0;
}

.year-label span {
  position: relative;
  background: white; /* match your background */
  padding: 0 0.5em;
  z-index: 1;
}


  /* author fields */
  .authors {
}
.authors .underline {
  text-decoration: underline;
}

  /* title field */
  .title {
  font-weight: bold;
}
  
</style>


<script>
  /* toggle */
function toggleBox(id) {
  const el = document.getElementById(id);
  if (el) el.classList.toggle('noshow');
}
</script>

<script type="text/javascript" async
  /* latex */
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>

<script>
  /* underline my name */
document.addEventListener("DOMContentLoaded", function() {
  document.querySelectorAll('.authors').forEach(el => {
    const authorName = "Niels Holtgrefe";
    const regex = new RegExp(`(${authorName})`, "g");

    // Replace only the name with the underlined span
    //el.innerHTML = el.innerHTML.replace(regex, '<span class="underline">$1</span>');
  });
});
</script>


You can also browse my <a href="https://scholar.google.com/citations?user=wJh7R3UAAAAJ" target="_blank">Google Scholar profile</a>. See the bottom 

<h2>Publications</h2>

<div class="year-label"><span>2025</span></div>

<ul class="no-bullets"> 

  <li><span class="pubnum">[5]</span>
    <span class="authors">Aviva K. Englander, Martin Frohn, Elizabeth Gross, Niels Holtgrefe, Leo van Iersel, Mark Jones, and Seth Sullivant.</span><br>
    <span class="title">Identifiability of Phylogenetic Level-2 Networks under the Jukes-Cantor Model.</span><br>
    <i>Preprint</i>, 2025. bioRxiv:10.1101/2025.04.18.649493.<br>

    <div class="button-group">
      <a class="link-button" href="https://www.biorxiv.org/content/10.1101/2025.04.18.649493v1" target="_blank">bioRiv</a>
      <button class="toggle-button" onclick="toggleBox('abs_englander2025identifiability')">abstract</button>
      <button class="toggle-button" onclick="toggleBox('bib_englander2025identifiability')">bibtex</button>
    </div>

    <div id="abs_englander2025identifiability" class="abstract-box noshow">
      We consider the fundamental question of which evolutionary histories can potentially be reconstructed from sufficiently long DNA sequences, by studying the identifiability of phylogenetic networks from data generated under Markov models of DNA evolution. This topic has previously been studied for phylogenetic trees and for phylogenetic networks that are level-1, which means that reticulate evolutionary events were restricted to be independent in the sense that the corresponding cycles in the network are non-overlapping.<br>
      In this paper, we study the identifiability of phylogenetic networks from DNA sequence data under Markov models of DNA evolution for more general classes of networks that may contain pairs of tangled reticulations. Our main result is generic identifiability, under the Jukes-Cantor model, of binary semi-directed level-2 phylogenetic networks that satisfy two additional conditions called triangle-free and strongly tree-child.<br>
      We also consider level-1 networks and show stronger identifiability results for this class than what was known previously. In particular, we show that the number of reticulations in a level-1 network is identifiable under the Jukes-Cantor model.<br>
      Moreover, we prove general identifiability results that do not restrict the network level at all and hold for the Jukes-Cantor as well as for the Kimura-2-Parameter model. We show that any two binary semi-directed phylogenetic networks are distinguishable if they do not display exactly the same 4-leaf subtrees, called quartets. This has direct consequences regarding the blobs of a network, which are its reticulated components. We show that the tree-of-blobs of a network, the global branching structure of the network, is always identifiable, as well as the circular ordering of the subnetworks around each blob, for networks in which edges do not cross and taxa are on the outside.    </div>

    <div id="bib_englander2025identifiability" class="bibtex noshow">
      <pre>
@misc{englander2025identifiability,
  author = {Englander, Aviva K and Frohn, Martin and Gross, Elizabeth and Holtgrefe, Niels and van Iersel, Leo and Jones, Mark and Sullivant, Seth},
  title = {Identifiability of Phylogenetic Level-2 Networks under the Jukes-Cantor Model},
  year={2025},
  archivePrefix={bioRxiv},
  eprint={10.1101/2025.04.18.649493}
}
      </pre>
    </div>
  </li>

  
  <li><span class="pubnum">[4]</span>
    <span class="authors">Martin Frohn, Niels Holtgrefe, Leo van Iersel, Mark Jones, and Steven Kelk.</span><br>
    <span class="title">Reconstructing semi-directed level-1 networks using few quarnets.</span><br>
    <i>Journal of Computer and System Sciences</i>, 152:103655, 2025. doi:10.1016/j.jcss.2025.103655.<br>

    <div class="button-group">
      <a class="link-button" href="https://doi.org/10.1016/j.jcss.2025.103655" target="_blank">journal</a>
      <a class="link-button" href="https://arxiv.org/abs/2409.06034" target="_blank">arXiv</a>
      <button class="toggle-button" onclick="toggleBox('abs_frohn2025reconstructing')">abstract</button>
      <button class="toggle-button" onclick="toggleBox('bib_frohn2025reconstructing')">bibtex</button>
    </div>

    <div id="abs_frohn2025reconstructing" class="abstract-box noshow">
Semi-directed networks are partially directed graphs that model evolution where the directed edges represent reticulate evolutionary events. We present an algorithm that reconstructs binary \(n\)-leaf semi-directed level-1 networks in \( O( n^2) \) time from its quarnets (4-leaf subnetworks). Our method assumes we have direct access to all quarnets, yet uses only an asymptotically optimal number of \( O(n \log n) \) quarnets. When the network is assumed to contain no triangles, our method instead relies only on four-cycle quarnets and the splits of the other quarnets. A variant of our algorithm works with quartets rather than quarnets and we show that it reconstructs most of a semi-directed level-1 network from an asymptotically optimal \( O(n \log n) \) of the quartets it displays. Additionally, we provide an \( O(n^3) \) time algorithm that reconstructs the tree-of-blobs of any binary \( n\)-leaf semi-directed network with unbounded level from \( O(n^3) \) splits of its quarnets.</div>

    <div id="bib_frohn2025reconstructing" class="bibtex noshow">
      <pre>
@article{frohn2025reconstructing,
  title={Reconstructing semi-directed level-1 networks using few quarnets},
  author={Frohn, Martin and Holtgrefe, Niels and van Iersel, Leo and Jones, Mark and Kelk, Steven},
journal={Journal of Computer and System Sciences},
doi={10.1016/j.jcss.2025.103655},
volume = {152},
pages = {103655},
year = {2025}
}
      </pre>
    </div>
  </li>


  
  <li><span class="pubnum">[3]</span>
    <span class="authors">Niels Holtgrefe, Katharina T. Huber, Leo van Iersel, Mark Jones, Samuel Martin, and Vincent Moulton.</span><br>
    <span class="title"><span style="font-variant: small-caps;">Squirrel</span>: Reconstructing Semi-directed Phylogenetic Level-1 Networks from Four-Leaved Networks or Sequence Alignments.</span><br>
    <i>Molecular Biology and Evolution</i>, 42(4):msaf067, 2025. doi:10.1093/molbev/msaf067.<br>

    <div class="button-group">
      <a class="link-button" href="https://doi.org/10.1093/molbev/msaf067" target="_blank">journal</a>
      <a class="link-button" href="https://www.biorxiv.org/content/10.1101/2024.11.01.621567v2" target="_blank">bioRxiv</a>
      <a class="link-button" href="https://github.com/nholtgrefe/squirrel" target="_blank">software</a>
      <button class="toggle-button" onclick="toggleBox('abs_holtgrefe2025squirrel')">abstract</button>
      <button class="toggle-button" onclick="toggleBox('bib_holtgrefe2025squirrel')">bibtex</button>
    </div>

    <div id="abs_holtgrefe2025squirrel" class="abstract-box noshow">
    With the increasing availability of genomic data, biologists aim to find more accurate descriptions of evolutionary histories influenced by secondary contact, where diverging lineages reconnect before diverging again. Such reticulate evolutionary events can be more accurately represented in phylogenetic networks than in phylogenetic trees. Since the root location of phylogenetic networks cannot be inferred from biological data under several evolutionary models, we consider semi-directed (phylogenetic) networks: partially directed graphs without a root in which the directed edges represent reticulate evolutionary events. By specifying a known outgroup, the rooted topology can be recovered from such networks. We introduce the algorithm <span style="font-variant: small-caps;">Squirrel</span> (<u>S</u>emi-directed <u>Qu</u>arnet-based <u>I</u>nfe<u>r</u>ence to <u>Re</u>construct <u>L</u>evel-1 Networks), which constructs a semi-directed level-1 network from a full set of quarnets (four-leaf semi-directed networks). Our method also includes a heuristic to construct such a quarnet set directly from sequence alignments. We demonstrate <span style="font-variant: small-caps;">Squirrel</span>'s performance through simulations and on real sequence datasets, the largest of which contains 29 aligned sequences close to 1.7 Mbp long. The resulting networks are obtained on a standard laptop within a few minutes. Lastly, we prove that <span style="font-variant: small-caps;">Squirrel</span> is combinatorially consistent: given a full set of quarnets coming from a triangle-free semi-directed level-1 network, it is guaranteed to reconstruct the original network. <span style="font-variant: small-caps;">Squirrel</span> is implemented in Python, has an easy-to-use graphical user interface that takes sequence alignments or quarnets as input, and is freely available at <a href="https://github.com/nholtgrefe/squirrel" target="_blank">https://github.com/nholtgrefe/squirrel</a>.
    </div>

    <div id="bib_holtgrefe2025squirrel" class="bibtex noshow">
      <pre>
@article{holtgrefe2025squirrel,
  title={Squirrel: Reconstructing semi-directed phylogenetic level-1 networks from four-leaved networks or sequence alignments},
  author={Holtgrefe, Niels and Huber, Katharina T and van Iersel, Leo and Jones, Mark and Martin, Samuel and Moulton, Vincent},
journal={Molecular Biology and Evolution},
doi={10.1093/molbev/msaf067},
volume = {42},
number = {4},
pages = {msaf067},
year = {2025}
}
      </pre>
    </div>
  </li>
</ul>

<div class="year-label"><span>2024</span></div>

<ul class="no-bullets"> 
  <li><span class="pubnum">[2]</span>
    <span class="authors">Martin Frohn, Niels Holtgrefe, Leo van Iersel, Mark Jones, and Steven Kelk.</span><br>
    <span class="title">Invariants for level-1 phylogenetic networks under the random walk 4-state Markov model.</span><br>
    <i>Preprint</i>, 2024. arXiv:2407.11720.<br>

    <div class="button-group">
      <a class="link-button" href="https://arxiv.org/abs/2407.11720" target="_blank">arXiv</a>
      <button class="toggle-button" onclick="toggleBox('abs_frohn2024invariants')">abstract</button>
      <button class="toggle-button" onclick="toggleBox('bib_frohn2024invariants')">bibtex</button>
    </div>

    <div id="abs_frohn2024invariants" class="abstract-box noshow">
    Phylogenetic networks can represent evolutionary events that cannot be described by phylogenetic trees, such as hybridization, introgression, and lateral gene transfer. Studying phylogenetic networks under a statistical model of DNA sequence evolution can aid the inference of phylogenetic networks. Most notably Markov models like the Jukes-Cantor or Kimura-3 model can been employed to infer a phylogenetic network using phylogenetic invariants. In this article we determine all quadratic invariants for sunlet networks under the random walk 4-state Markov model, which includes the aforementioned models. Taking toric fiber products of trees and sunlet networks, we obtain a new class of invariants for level-1 phylogenetic networks under the same model. Furthermore, we apply our results to the identifiability problem of a network parameter. In particular, we prove that our new class of invariants of the studied model is not sufficient to derive identifiability of quarnets (4-leaf networks). Moreover, we provide an efficient method that is faster and more reliable than the state-of-the-art in finding a significant number of invariants for many level-1 phylogenetic networks.    
    </div>

    <div id="bib_frohn2024invariants" class="bibtex noshow">
      <pre>
@misc{frohn2024invariants,
  author = {Frohn, Martin and Holtgrefe, Niels and van Iersel, Leo and Jones, Mark and Kelk, Steven},
  title = {Invariants for level-1 phylogenetic networks under the random walk 4-state Markov model},
  year={2024},
  archivePrefix={arXiv},
  eprint={2407.11720}
}
      </pre>
    </div>
  </li>


  
  <li><span class="pubnum">[1]</span>
    <span class="authors">Niels Holtgrefe, Leo van Iersel, and Mark Jones.</span><br>
    <span class="title">Exact and Heuristic Computation of the Scanwidth of Directed Acyclic Graphs.</span><br>
    <i>Preprint</i>, 2024. arXiv:2403.12734.<br>

    <div class="button-group">
      <a class="link-button" href="https://arxiv.org/abs/2403.12734" target="_blank">arXiv</a>
      <a class="link-button" href="https://github.com/nholtgrefe/scanwidth" target="_blank">software</a>
      <button class="toggle-button" onclick="toggleBox('abs_holtgrefe2024exact')">abstract</button>
      <button class="toggle-button" onclick="toggleBox('bib_holtgrefe2024exact')">bibtex</button>
    </div>

    <div id="abs_holtgrefe2024exact" class="abstract-box noshow">
  To measure the tree-likeness of a directed acyclic graph (DAG), a new width parameter that considers the directions of the arcs was recently introduced:
  <em>scanwidth</em>. We present the first algorithm that efficiently computes the exact scanwidth of general DAGs. For DAGs with one root and scanwidth
  \( k \), it runs in \( O(k \cdot n^k \cdot m) \) time. The algorithm also functions as an FPT algorithm with complexity
  \( O(2^{4\ell - 1} \cdot \ell \cdot n + n^2) \) for phylogenetic networks of level-\( \ell \), a type of DAG used to depict evolutionary
  relationships among species. Our algorithm performs well in practice, being able to compute the scanwidth of synthetic networks up to 30 reticulations
  and 100 leaves within 500 seconds. Furthermore, we propose a heuristic that obtains an average practical approximation ratio of 1.5 on these networks.
  While we prove that the scanwidth is bounded from below by the treewidth of the underlying undirected graph, experiments suggest that for networks
  the parameters are close in practice.    
    </div>

    <div id="bib_holtgrefe2024exact" class="bibtex noshow">
      <pre>
@misc{holtgrefe2024exact,
  author = {Holtgrefe, Niels and van Iersel, Leo and Jones, Mark},
  title = {Exact and Heuristic Computation of the Scanwidth of Directed Acyclic Graphs},
  year={2024},
  archivePrefix={arXiv},
  eprint={2403.12734}
}
      </pre>
    </div>
  </li>

  
</ul>

